python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -0,0 +1,1219 @@
+        isCalibration=True,
+        minimum=0,  # can use either bfKernel or newBFKernel
+    )
+    dark = cT.PrerequisiteInput(
+        name='dark',
+        doc="Input dark calibration.",
+        storageClass="ExposureF",
+        dimensions=["instrument", "detector"],
+        isCalibration=True,
+    )
+    outputExposure = cT.Output(
+        name='postISRCCD',
+        doc="Output ISR processed exposure.",
+        storageClass="Exposure",
+        dimensions=["instrument", "exposure", "detector"],
+    )
+    preInterpExposure = cT.Output(
+        name='preInterpISRCCD',
+        doc="Output ISR processed exposure, with pixels left uninterpolated.",
+        storageClass="ExposureF",
+        dimensions=["instrument", "exposure", "detector"],
+    )
+    outputStatistics = cT.Output(
+        name="isrStatistics",
+        doc="Output of additional statistics table.",
+        storageClass="StructuredDataDict",
+        dimensions=["instrument", "exposure", "detector"],
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+        if config.doDiffNonLinearCorrection is not True:
+            self.prerequisiteInputs.remove("dnlLUT")
+        if config.doBias is not True:
+            self.prerequisiteInputs.remove("bias")
+        if config.doDeferredCharge is not True:
+            self.prerequisiteInputs.remove("deferredChargeCalib")
+        if config.doLinearize is not True:
+            self.prerequisiteInputs.remove("linearizer")
+        if config.usePtcGains is not True and config.usePtcReadNoise is not True:
+            self.prerequisiteInputs.remove("ptc")
+        if config.doCrosstalk is not True:
+            self.prerequisiteInputs.remove("crosstalkSources")
+            self.prerequisiteInputs.remove("crosstalk")
+        if config.doDefect is not True:
+            self.prerequisiteInputs.remove("defects")
+        if config.doBrighterFatter is not True:
+            self.prerequisiteInputs.remove("bfKernel")
+        if config.doDark is not True:
+            self.prerequisiteInputs.remove("dark")
+
+        if config.doWrite is not True:
+            self.outputs.remove("outputExposure")
+            self.outputs.remove("preInterpExposure")
+
+        if config.doSaveInterpPixels is not True:
+            self.outputs.remove("preInterpExposure")
+
+        if config.doCalculateStatistics is not True:
+            self.outputs.remove("outputStatistics")
+
+
+class IsrTaskLSSTConfig(pipeBase.PipelineTaskConfig,
+                        pipelineConnections=IsrTaskLSSTConnections):
+    """Configuration parameters for IsrTaskLSST.
+
+    Items are grouped in the order in which they are executed by the task.
+    """
+    datasetType = pexConfig.Field(
+        dtype=str,
+        doc="Dataset type for input data; users will typically leave this alone, "
+        "but camera-specific ISR tasks will override it.",
+        default="raw",
```

Comment:
I'm pretty sure this is a hangover from trying to support certain types of pre-processed decam images from gen2.  We do not want this configuration at all.  If, strangely, you need a different dataset type then the `ccdExposure` connection name would be the thing to override).